ci/ci/github.py

```diff
@@ -527,6 +546,7 @@ async def _start_build(self, db: Database, batch_client):
             self.set_build_state('error')
             self.source_sha_failed = True
             self.target_branch.state_changed = True
+            await release_namespace(db, namespace)
```

Comment:
Should we also release the namespace if there's a CancelledError? I think I don't fully understand how namespaces are resource managed.